Add code reviews most common errors (#21)
Harmos274/SQA-Assignment: README.md

### How to make a code review ?

### Most common errors

> We will address **4 common errors** that are already encountered while doing a code review. These can be avoid by following good practices. You will find [here](https://www.backhub.co/blog/best-practices-reviewing-pull-requests-github) and [here](https://smartbear.com/learn/code-review/best-practices-for-peer-code-review/) an article which you could use as a guide for your code reviews.

Reviewer: ```suggestion
We will address **4 common errors** that we already encountered during code reviews. Those could have been avoided by following some good practices. You will find [here](https://www.backhub.co/blog/best-practices-reviewing-pull-requests-github) and [here](https://smartbear.com/learn/code-review/best-practices-for-peer-code-review/) two articles which you can use as guides when you're reviewing code.
```

#### Do NOT let the code cool down

If one of your colleague ask you to review its work. He will be certainly waiting for your review in the next minutes and then will switch to something else. Thus when you will do the review, maybe few hours later he will have to switch back to his previous work which will have an **extra switching cost** for him. Moreover, he could have forget some aspect of his code he will have to remind in order to fix its code.

#### Provide a constructive feedback

What is worse than receiving negative and inaccurate feedback such as :

 > WTF did you do ? Why don't you use only X instead?!

We engage you to use as much as possible affirmative form, using sentences begining with :

> "I would suggest you to..."

or

> "I am suprised by... I would rather use..."

> At the same time, do not hesitate to share any kind of positive feeling you could havelooking at the code. Congratulation are always welcome !

Reviewer: ```suggestion
At the same time, do not hesitate to share any kind of positive feeling you could have when you're reviewing the code. Congratulation are always welcome !
```

#### Keep your ego out

This is probably one of the most common mistake. If you find yourself in a position where you are reviewing someone who is senior to you, **NEVER** assume that you have nothing to learn from him. To the contrary, always assume that this person can teach you something that you don't know yet. Especially if your are facing a very young developer (such as an intern), they are more enclined than to find unexpected solutions.
They succeed in crossing the crocodile river swimming since no one told them that it was impossible.

#### Never assume that the code will launch

> A very common mistake is to review a code, assuming that the code you are reviewing is working as intended. **ALWAYS** run the code on your own to check everything works as expected. The last thing you want is to face an emergency while production is broken due to a piece of code you reviewed.

Reviewer: ```suggestion
A very common mistake is to review a code, assuming that the code you are reviewing is working as intended. **ALWAYS** run the code on your own to check that everything works as expected. The last thing you want is to face an emergency while production is broken due to a piece of code you reviewed.
```
